language-support/hs/bindings/src/LedgerHL.hs

```diff
@@ -0,0 +1,188 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module LedgerHL(
+    Port(..),
+    LedgerHandle,
+    LedgerId,
+    Party(..),
+    Transaction,
+    ResponseStream,
+    connect,
+    identity,
+    getTransactionStream,
+    nextResponse,
+    ) where
+
+import           Control.Concurrent
+import           Control.Monad.Fix  (fix)
+import qualified Data.Map           as Map
+import           Data.Text.Lazy     (Text)
+import qualified Data.Text.Lazy     as Text
+import           Prelude            hiding (log)
+
+import           LedgerId
+import           Transaction
+
+import Network.GRPC.HighLevel.Generated (
+    withGRPCClient,
+    ClientRequest(
+            ClientReaderRequest,
+            ClientNormalRequest
+            ),
+    ClientResult(
+            ClientErrorResponse,
+            ClientReaderResponse,
+            ClientNormalResponse
+            ),
+    MetadataMap(..),
+    GRPCMethodType(Normal,ServerStreaming),
+    ClientConfig(..),
+    Port(..),
+    Host(..),
+    )
+
+import Com.Digitalasset.Ledger.Api.V1.LedgerIdentityService as LIS
+import Com.Digitalasset.Ledger.Api.V1.TransactionService as TS
+
+import Com.Digitalasset.Ledger.Api.V1.TransactionService(GetTransactionsRequest(..))
+import Com.Digitalasset.Ledger.Api.V1.TransactionFilter(TransactionFilter(..))
+import Com.Digitalasset.Ledger.Api.V1.TransactionFilter(Filters(..))
+import Com.Digitalasset.Ledger.Api.V1.LedgerOffset(LedgerOffset(..))
+import Com.Digitalasset.Ledger.Api.V1.LedgerOffset(LedgerOffsetValue(..))
+import Com.Digitalasset.Ledger.Api.V1.LedgerOffset(LedgerOffset_LedgerBoundary(..))
+import Com.Digitalasset.Ledger.Api.V1.TraceContext(TraceContext)
+import qualified Proto3.Suite.Types as PST
+
+
+data LedgerHandle = LedgerHandle { port :: Port, id :: LedgerId }
+
+identity :: LedgerHandle -> LedgerId
+identity LedgerHandle{id} = id
+
+
+newtype Party = Party { party :: Text }
+
+instance Show Party where
+    show Party{party} = Text.unpack party
+    
+
+newtype ResponseStream a = ResponseStream { chan :: Chan a }
+
+nextResponse :: ResponseStream a -> IO a
+nextResponse ResponseStream{chan} = readChan chan
+
+
+connect :: Port -> IO LedgerHandle
+connect port = do
+    id <- getLedgerIdentity port
+    return$ LedgerHandle {port, id}
+
+getLedgerIdentity :: Port -> IO LedgerId
+getLedgerIdentity port = do
+    log$ "ledgerIdentity"
```

Comment:
This line fails the build; should be
```suggestion
    log "ledgerIdentity"
```
according to the linter

Reply:
yup